src/main/java/de/tum/in/www1/artemis/web/rest/ModelingSubmissionResource.java

```diff
@@ -198,29 +204,34 @@ public ModelingSubmissionResource(SubmissionRepository submissionRepository, Res
             return forbidden();
         }
 
-        // load submission with results either by resultId or by correctionRound
-        if (resultId != null) {
-            // load the submission with additional needed properties
-            modelingSubmission = (ModelingSubmission) submissionRepository.findOneWithEagerResultAndFeedback(submissionId);
-            // check if result exists
-            Result result = modelingSubmission.getManualResultsById(resultId);
-            if (result == null) {
-                return ResponseEntity.badRequest()
-                        .headers(HeaderUtil.createFailureAlert(applicationName, true, "ModelingSubmission", "ResultNotFound", "No Result was found for the given ID.")).body(null);
+        if (!withoutResults) {
+            // load submission with results either by resultId or by correctionRound
+            if (resultId != null) {
+                // load the submission with additional needed properties
+                modelingSubmission = (ModelingSubmission) submissionRepository.findOneWithEagerResultAndFeedback(submissionId);
+                // check if result exists
+                Result result = modelingSubmission.getManualResultsById(resultId);
+                if (result == null) {
+                    return ResponseEntity.badRequest()
+                            .headers(HeaderUtil.createFailureAlert(applicationName, true, "ModelingSubmission", "ResultNotFound", "No Result was found for the given ID."))
+                            .body(null);
+                }
+            }
+            else {
+                // load and potentially lock the submission with additional needed properties by correctionRound
+                modelingSubmission = modelingSubmissionService.lockAndGetModelingSubmission(submissionId, modelingExercise, correctionRound);
             }
-        }
-        else {
-            // load and potentially lock the submission with additional needed properties by correctionRound
-            modelingSubmission = modelingSubmissionService.lockAndGetModelingSubmission(submissionId, modelingExercise, correctionRound);
         }
 
         // Make sure the exercise is connected to the participation in the json response
         studentParticipation.setExercise(modelingExercise);
         modelingSubmission.getParticipation().getExercise().setGradingCriteria(gradingCriteria);
 
-        // prepare modelingSubmission for response
-        modelingSubmissionService.hideDetails(modelingSubmission, user);
-        modelingSubmission.removeNotNeededResults(correctionRound, resultId);
+        if (!withoutResults) {
+            // prepare modelingSubmission for response
+            modelingSubmissionService.hideDetails(modelingSubmission, user);
```

Comment:
Shouldn't you still hide details even without the results?

Reply:
definitely, thanks!